community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/input/InputEntityCacherTest.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.input;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.RuleChain;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.neo4j.io.fs.StoreChannel;
+import org.neo4j.kernel.impl.store.format.RecordFormat;
+import org.neo4j.kernel.impl.store.format.RecordFormats;
+import org.neo4j.test.RandomRule;
+import org.neo4j.test.Randoms;
+
+import static java.lang.Math.abs;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class InputEntityCacherTest
+{
+
+    private static final AtomicInteger uniqueIdGenerator = new AtomicInteger( 10 );
+    private final ExpectedException expectedException = ExpectedException.none();
+    private final RandomRule randomRule = new RandomRule();
+    @Rule
+    public RuleChain ruleChain = RuleChain.outerRule( randomRule ).around( expectedException );
+
+    @Test
+    public void notAllowCreationOfUnsupportedNumberOfProperties() throws IOException
+    {
+        initExpectedException();
+        cacheNodeWithProperties(10, 10);
```

Comment:
Code style.
Magic numbers would be fine if not used in the expected exception message, (`initExpectedException()`).
I would expect `cacheNodeWithProperties( 10, 10 )` to pass and `cacheNodeWithProperties( 11, 10 )` to fail.